shifthelper_webinterface/log.py

```diff
@@ -0,0 +1,54 @@
+from flask import current_app
+from time import sleep
+from itertools import count
+
+
+def get_size(f):
+    pos = f.tell()
+    f.seek(0, 2)
+    size = f.tell()
+    f.seek(pos)
+    return size
+
+
+def log_generator(max_size=50000):
+    '''
+    Generates the bodies of server sent events from the shifthelper log.
+    To be used as streamed response use flask.Respone(log_generator()).
+    Can handle logfile rotation by comparing the size of the file
+    to the current position.
+    '''
+    with open(current_app.config['shifthelper_log'], 'rb') as f:
+        for i in count():
+            size = get_size(f)
+
+            # truncate to max_size
+            if i == 0 and size > max_size:
+                f.seek(-max_size, 2)
+                yield build_sse('{:*^40}'.format(
+                    'Log output truncated to {} kB'.format(max_size // 1000)
+                ), i)
+
+            if f.tell() > size:
+                f.seek(0)
+                msg = build_sse('{:*^40}'.format(
+                    'Log file was truncated, probably log rotation'
+                ), i)
+            else:
+                msg = f.read().decode()
+
+            yield build_sse(msg, i)
+            sleep(1)
+
+
+def build_sse(message, id_=None):
```

Comment:
I do not know what `sse` stands for ... (and I was also unable to find it by quickly googling ... https://en.wikipedia.org/wiki/SSE)

maybe this name could be refactored ... unless it is a fixed term one simply has to learn.

Reply:
The full term is explained in the docstring and its the second in the list of the wikipedia article you linked.